File: reverse-string/reverse-string.go
func reverseString(s []byte)  {
    // Recursive solution
    var swap func(int, int)
    swap = func(i, j int) {
        if i<j {
            s[i], s[j] = s[j], s[i]
            i++
            j--
            swap(i, j)
        }
    }
    i, j := 0, len(s)-1
    swap(i, j)
}

File: reverse-string/reverse-string.py
class Solution:
    def reverseString(self, s: List[str]) -> None:
        temp = s[::-1]
        # for some reason, this won't let me do `s = temp`
        # or `s = temp[:]`
        s.clear()
        for i in temp:
            s.append(i)
        